src/EnergyPlus/CommandLineInterface.cc

```diff
@@ -95,163 +94,183 @@ namespace CommandLineInterface {
 
-        if (!opt.isSet("-i") && !legacyMode) {
-            state.dataStrGlobals->inputIddFilePath = state.dataStrGlobals->exeDirectoryPath / state.dataStrGlobals->inputIddFilePath;
-        }
+        auto *annualOpt = app.add_flag("-a,--annual", state.dataGlobal->AnnualSimulation, "Force annual simulation");
 
-        {
-            std::string outDirPathName;
-            opt.get("-d")->getString(outDirPathName);
-            state.dataStrGlobals->outDirPath = fs::path{outDirPathName};
-        }
-
-        state.dataGlobal->runReadVars = opt.isSet("-r");
-
-        state.dataGlobal->DDOnlySimulation = opt.isSet("-D");
-
-        state.dataGlobal->AnnualSimulation = opt.isSet("-a");
-
-        state.dataGlobal->outputEpJSONConversion = opt.isSet("-c");
-
-        state.dataGlobal->outputEpJSONConversionOnly = opt.isSet("--convert-only");
-
-        bool eplusRunningViaAPI = state.dataGlobal->eplusRunningViaAPI;
+        app.add_flag("-D,--design-day", state.dataGlobal->DDOnlySimulation, "Force design-day-only simulation")->excludes(annualOpt);
 
-        opt.get("-j")->getInt(state.dataGlobal->numThread);
+        app.add_option("-d,--output-directory", state.dataStrGlobals->outDirPath, "Output directory path (default: current directory)")
+            ->option_text("DIR")
+            ->required(false);
+        // ->check(CLI::ExistingDirectory) // We don't require it to exist, we make it if needed
 
-        if (state.dataGlobal->numThread == 0) {
-            DisplayString(state, "Invalid value for -j arg. Defaulting to 1.");
-            state.dataGlobal->numThread = 1;
-        } else if (state.dataGlobal->numThread > (int)std::thread::hardware_concurrency()) {
-            DisplayString(state,
-                          fmt::format("Invalid value for -j arg. Value exceeds num available. Defaulting to num available. -j {}",
-                                      (int)std::thread::hardware_concurrency()));
-            state.dataGlobal->numThread = (int)std::thread::hardware_concurrency();
-        }
-
-        // Process standard arguments
-        if (opt.isSet("-h")) {
-            DisplayString(state, usage);
+        if (legacyMode) {
+            state.dataStrGlobals->inputIddFilePath = "Energy+.idd";
+        } else {
+            state.dataStrGlobals->inputIddFilePath = state.dataStrGlobals->exeDirectoryPath / "Energy+.idd";
+        }
+        app.add_option(
+               "-i,--idd", state.dataStrGlobals->inputIddFilePath, "Input data dictionary path (default: Energy+.idd in executable directory)")
+            ->required(false)
+            ->option_text("IDD")
+            ->check(CLI::ExistingFile);
+
+        bool runEPMacro = false;
+        app.add_flag("-m,--epmacro", runEPMacro, "Run EPMacro prior to simulation");
+
+        std::string prefixOutName = "eplus";
+        app.add_option("-p,--output-prefix", prefixOutName, "Prefix for output file names (default: eplus)")->required(false)->option_text("PRE");
```

Comment:
Here's an example option. It pretty clear what's going on. The default `std::string prefixOutName` is set to "eplus", then it's reassigned by CLI11 **if** the non required option is passed.